src/hera/workflow.py

```diff
@@ -264,6 +264,9 @@ def create(self) -> "Workflow":
         if self.in_context:
             raise ValueError("Cannot invoke `create` when using a Hera context")
 
+        if self.service is None:
+            self.service = WorkflowService()
+
         self.service.create_workflow(self.build())
```

Comment:
Looks like a great start @nikhiljha! Thank you for the PR! I think there might be other places to update as well. Specifically, the `CronWorkflow` and `WorkflowTemplate`. Can the PR include those as well?

Reply:
@flaviuvadan Ah okay! Just pushed a new PR that should fix it anywhere (including in future objects), since they're all subclassed on `Workflow`.